### ISSUE TITLE
gem list includes a " ruby" next to the version for some of the default gems

### ISSUE THREAD

**Issue by eregon** [rubygems/rubygems#8178]
### Describe the problem as clearly as you can

`gem list` includes ` ruby` after the version for some default gems but not others.
For instance abbrev base64 benchmark have it below but no other gems have that (even though there are plenty of other pure-Ruby default gems like drb or csv).

```
$ ruby -v
ruby 3.3.5 (2024-09-03 revision ef084cc8f4) [x86_64-linux]
$ gem list

*** LOCAL GEMS ***

abbrev (default: 0.1.2 ruby)
base64 (default: 0.2.0 ruby)
benchmark (default: 0.3.0 ruby)
bigdecimal (default: 3.1.5)
bundler (default: 2.5.16, 2.4.19)
cgi (default: 0.4.1)
concurrent-ruby (1.3.4)
csv (default: 3.2.8)
date (default: 3.3.4)
debug (1.9.2, 1.9.1)
delegate (default: 0.3.1)
did_you_mean (default: 1.6.3)
digest (default: 3.1.1)
drb (default: 2.2.0)
english (default: 0.8.0)
erb (default: 4.0.3)
error_highlight (default: 0.6.0)
etc (default: 1.4.3)
fcntl (default: 1.1.0)
fiddle (default: 1.1.2)
fileutils (default: 1.7.2)
find (default: 0.2.0)
forwardable (default: 1.3.3)
getoptlong (default: 0.2.1)
gruff (0.25.0)
histogram (0.2.4.1)
io-console (default: 0.7.1)
io-nonblock (default: 0.3.0)
io-wait (default: 0.3.1)
ipaddr (default: 1.2.6)
irb (default: 1.13.1)
json (2.7.2, default: 2.7.1)
language_server-protocol (3.17.0.3)
logger (default: 1.6.0)
matrix (0.4.2)
minitest (5.20.0)
mutex_m (default: 0.2.0)
net-ftp (0.3.4)
net-http (default: 0.4.1)
net-imap (0.4.9.1)
net-pop (0.1.2)
net-protocol (default: 0.2.2)
net-smtp (0.4.0.1)
nkf (default: 0.1.3)
observer (default: 0.1.2)
open-uri (default: 0.4.1)
open3 (default: 0.2.1)
openssl (default: 3.2.0)
optparse (default: 0.4.0)
ostruct (default: 0.6.0)
path (2.1.0)
pathname (default: 0.3.0)
pkg-config (1.5.6)
power_assert (2.0.3)
pp (default: 0.5.0)
prettyprint (default: 0.2.0)
prime (0.1.2)
prism (default: 0.19.0)
pstore (default: 0.1.3)
psych (default: 5.1.2)
racc (1.7.3)
rake (13.1.0)
rbs (3.4.0)
rdoc (default: 6.6.3.1)
readline (default: 0.0.4)
reline (default: 0.5.7)
resolv (default: 0.3.0)
resolv-replace (default: 0.1.1)
rexml (3.3.6)
rinda (default: 0.2.0)
rmagick (6.0.1)
rss (0.3.1)
ruby-lsp (0.17.9)
ruby2_keywords (default: 0.0.5)
securerandom (default: 0.3.1)
set (default: 1.1.0)
shellwords (default: 0.2.0)
singleton (default: 0.2.0)
sorbet-runtime (0.5.11495)
stringio (default: 3.1.1)
strscan (default: 3.0.9)
syntax_suggest (default: 2.0.0)
syslog (default: 0.1.2)
tempfile (default: 0.2.1)
test-unit (3.6.1)
time (default: 0.3.0)
timeout (default: 0.4.1)
tmpdir (default: 0.2.0)
tsort (default: 0.2.0)
typeprof (0.21.9)
un (default: 0.3.0)
uri (default: 0.13.1)
weakref (default: 0.1.3)
yaml (default: 0.3.0)
zlib (default: 3.1.1)
```

### Did you try upgrading RubyGems?

No, I would like to avoid altering this installation of CRuby.
I have checked on ruby master and also `3.4.0preview2` and it seems to not reproduce there.
I'm not sure this means the bug is fixed either, it might be transient, cc @andrykonchin who saw it a few times but not always.
It would be great if it can be fixed for the next Ruby 3.3.x release.

### Run `gem env` and paste the output below

```
RubyGems Environment:
  - RUBYGEMS VERSION: 3.5.16
  - RUBY VERSION: 3.3.5 (2024-09-03 patchlevel 100) [x86_64-linux]
  - INSTALLATION DIRECTORY: /home/eregon/.rubies/ruby-3.3.5/lib/ruby/gems/3.3.0
  - USER INSTALLATION DIRECTORY: /home/eregon/.gem/ruby/3.3.0
  - RUBY EXECUTABLE: /home/eregon/.rubies/ruby-3.3.5/bin/ruby
  - GIT EXECUTABLE: /usr/bin/git
  - EXECUTABLE DIRECTORY: /home/eregon/.rubies/ruby-3.3.5/bin
  - SPEC CACHE DIRECTORY: /home/eregon/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /home/eregon/.rubies/ruby-3.3.5/etc
  - RUBYGEMS PLATFORMS:
     - ruby
     - x86_64-linux
  - GEM PATHS:
     - /home/eregon/.rubies/ruby-3.3.5/lib/ruby/gems/3.3.0
     - /home/eregon/.gem/ruby/3.3.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => true
     - :bulk_threshold => 1000
     - "gem" => "--no-document"
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /home/eregon/.rubies/ruby-3.3.5/bin
     - /home/eregon/bin
     - /usr/share/Modules/bin
     - /usr/local/bin
     - /usr/local/sbin
     - /usr/bin
     - /usr/sbin
     - /home/eregon/code/mx
     - /home/eregon/code/mspec/bin
     - /opt/crystal/bin
     - /home/eregon/.cargo/bin
```

**Comment by deivid-rodriguez** [comment]
This is fixed in the latest version of RubyGems, as you have observed. Closing!

**Comment by eregon** [comment]
Thanks, could you link to the fix?
Also will it be fixed in CRuby 3.3.6?

**Comment by deivid-rodriguez** [comment]
I don't recall the fix, but I recall I fixed it. It should be one of 3.5.17, 3.5.18, 3.5.19, 3.5.20, 3.5.21 or 3.5.22. I suggest you bisect it.

**Comment by deivid-rodriguez** [comment]
>  Also will it be fixed in CRuby 3.3.6?

I don't manage Ruby releases, but they usually include the latest Bundler & RubyGems so I'd say yes.
